docs/concepts/mixer.md

```diff
@@ -7,40 +7,210 @@ title: Mixer
+
+    message MetricDescriptor {
+      string name = 1;
+      string display_name = 2;
+      string description = 3;
+    
+      enum MetricKind {
+        METRIC_KIND_UNSPECIFIED = 0;
+        GAUGE = 1;
+        COUNTER = 2;
+      }
+    
+      MetricKind kind = 4;
+      ValueType value = 5;
+      repeated LabelDescriptor labels = 6;
+    }
+
+Within a deployment configuration, you can declare a metric descriptor using a snippet of YAML:
+
+    metrics:
+      - name: request_count
+        kind: 2 # COUNTER
+        value: 2 # INT64
+        description: request count by source, target, service, and code
+        labels:
+        - name: source
+          value_type: 1 # STRING
+        - name: target
+          value_type: 1 # STRING
+        - name: service
+          value_type: 1 # STRING
+        - name: method
+          value_type: 1 # STRING
+        - name: response_code
+          value_type: 2 # INT64
+
+This is declaring a descriptor called request_count. Because of the kind and value fields, policy objects associated with this
+descriptor represent 64-bit integer counters. Additionally, each policy object is uniquely identified via the 5 listed labels.
+Producing a policy object for such a descriptor requires 6 pieces of information:
+
+- a 64-bit integer metric value 
+- a source string
+- a target string
+- a service string
+- a method string
+- a 64-bit integer response code
+
+Here is an example snippet of Istio configuratin which produces a policy object for the above descriptor:
+
+      metrics:
+      - descriptor_name: request_count
+        value: "1"
+        labels:
+          source: source.name | "unknown"
+          target: target.name | "unknown"
+          service: api.name | "unknown"
+          method: api.method | "unknown"
+          response_code: response.code | 200
+
+Many such policy objects are created as part of attribute processing and they ultimately serve as input to
+adapters.
+
+Explicitly defining descriptors and creating policy objects for them is akin to types and objects in a traditional
+programming language. Doing so enables a few important scenarios:
+
+- Having the set of descriptors explicitly defined enables Istio to program backend systems to accept traffic produced
+by the mixer. For example, a MetricDescriptor provides all the information needed to program a backend system to accept metrics
+that conform to the descriptor's shape (it's value type and its set of labels).
+
+- It enables type checking of the deployment's configuration. Since attributes have strong types, and so do descriptors,
+Istio can provide a number of strong correctness guarantees of the system's configuration. Basically, if a chunk of
+configuration is accepted into the Istio system, it means the configuration passes a minimum correctness bar. Again, this
+plays the same role as types in a programming language.
+
+- It enables us to provide a strongly-typed scripting environment.
+
+## Scripting
```

Comment:
Again, I really don't think we want this yet. We don't support it all currently, and haven't figured out exactly how it is going to work, etc.